sympy/core/tests/test_assumptions.py

```diff
@@ -785,6 +797,7 @@ def test_Add_is_pos_neg():
     assert (p + r).is_extended_positive is None
 
 
+@XFAIL
 def test_Add_is_imaginary():
     nn = Dummy(nonnegative=True)
     assert (I*nn + I).is_imaginary  # issue 8046, 17
```

Comment:
This is XFAILed for now. It could be possible to improve the handling here but it isn't trivial. Really the root cause of a lot of difficulties around `is_imaginary` is that the definition is problematic because it excludes zero. It would be a lot easier to work with a definition that just says that an imaginary number is a complex number with zero real part. Then the set of imaginary numbers would be closed under addition making an inference like the one above trivial.

Reply:
Yes, although at least in the SymPy codebase the situation is more like the other way round: there are lots of places that try to work around the fact that zero is not considered imaginary e.g. the `(I*expr).is_extended_real` check that is removed in this PR.

Reply:
Relevant issue https://github.com/sympy/sympy/issues/7649

Reply:
Not sure why, but that link doesn't click through for me -- I have to ctrl+click to open new tab in Chrome

Reply:
The above test passes with the modification applied that I gave above (and is also added to the `_eval_is_imaginary` routine).
```python
>>> nn = Dummy(nonnegative=True)
>>> (I*nn + I).is_imaginary
True
```